### PR TITLE
Shorten `_get_frame_without_data()`

### PR DESCRIPTION
**Pull request by eerovaher**
### Description

The rewritten function is shorter and also less indented, so ignoring indentation-only changes can be helpful for reviewing:
```
$ git diff --stat --ignore-space-change main 
 astropy/coordinates/sky_coordinate_parsers.py | 70 +++++++++++++++++---------------------------------
 1 file changed, 24 insertions(+), 46 deletions(-)
```

- [x] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: astropy/coordinates/sky_coordinate_parsers.py [hunk 1]
```diff
@@ -91,55 +91,49 @@ def _get_frame_without_data(args, kwargs):
...
-                    raise ValueError(
-                        _conflict_err_msg.format(
-                            attr, getattr(frame, attr), kwargs[attr], "SkyCoord"
-                        )
+    if isinstance(frame, SkyCoord):
+        # If the frame was passed as a SkyCoord, we also want to preserve
+        # any extra attributes (e.g., obstime) if they are not already
+        # specified in the kwargs. We preserve these extra attributes by
+        # adding them to the kwargs dict:
+        for attr in frame._extra_frameattr_names:
```
Comment (mhvk) [inline astropy/coordinates/sky_coordinate_parsers.py:99]: Could avoid duplicating the `getattr` (and perhaps clarify the logic) by
```
value = getattr(frame, attr)
if attr not in kwargs:
    kwargs[attr] = value
elif np.any(value != kwargs[attr]):
    raise ValueError(...)
```

**Review by mhvk — APPROVED** [review]
This looks like a good simplification. I have just one further suggestion (but fine to ignore).

File: astropy/coordinates/sky_coordinate_parsers.py (hunk 1 above)
Comment (eerovaher) [reply astropy/coordinates/sky_coordinate_parsers.py:99]: Applying this suggestion indeed made the code less verbose.

**Review by mhvk — APPROVED** [review]
Thanks! Let's get it in...

**Pull request merged**